### PR TITLE
update to go 1.17

### PR DESCRIPTION
**Pull request by Dieterbe**
let's see if this works...

### CODE REVIEW

**Comment by Dieterbe** [comment]
@shanson7 or @fkaleo is either of you interested in reviewing/merging this?

File: scripts/qa/go-generate.sh [hunk 1]
```diff
@@ -17,20 +23,8 @@ if [ "$1" != "-f" -a "$1" != "--force" ]; then
...
-# lock the msgp tool to the same version as the vendored code, so the generated code is definitely compatible.
-# the generated code also tends to receive small tweaks over time, we wouldn't want our build to suddenly break
-# when that happens
-cd $gopath/src/github.com/tinylib/msgp
-git checkout $msgpRev
-go install
+go install golang.org/x/tools/cmd/stringer@$stringerVersion
+go install github.com/tinylib/msgp@$msgpVersion
 
 cd -
```
Comment (shanson7) [inline scripts/qa/go-generate.sh:29]: Do we still want `cd -` if L#31 (`cd $gopath/src/github.com/tinylib/msgp`) was removed?

Comment (Dieterbe) [reply scripts/qa/go-generate.sh]: good eye! pushed the fix.

**Comment by Dieterbe** [comment]
can i get an apprrrrrrove ? happy holidays !

**Review by robert-milan — APPROVED** [review]

**Pull request merged**